### PR TITLE
Added the ExtremeValueCopula class

### PR DESCRIPTION
**Pull request by regislebrun**
This class allows to define the most general 2D extreme value copulas through their Pickand function (A convex function A from [0,1] into [1/2, 1] such that max(t,1-t)<=A(t)<=1). The properties of the function are checked pointwise, see the documentation to know how to activate/deactivate the check and how to tune it.

### CODE REVIEW

File: lib/src/Uncertainty/Distribution/ExtremeValueCopula.cxx [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+  if (check)
+  {
+    // Check if max(t, 1-t) <= A(t) <= 1
+    const UnsignedInteger n = ResourceMap::GetAsUnsignedInteger("ExtremeValueCopula-CheckGridSize");
+    for (UnsignedInteger i = 0; i < n; ++i)
+    {
+      const Scalar t = i / (n - 1.0);
+      const Scalar valueF(pickandFunction(Point(1, t))[0]);
+      const Scalar valueH(pickandFunction.hessian(Point(1, t))(0, 0, 0));
+      if (!((valueF <= 1.0) && (valueF > std::max(t, 1.0 - t))))
```
Comment (dbarbier) [inline lib/src/Uncertainty/Distribution/ExtremeValueCopula.cxx]: According to the commit log message, should be: `valueF >= std::max(t, 1.0 - t)`

**Comment by regislebrun** [comment]
right, fixed.

File: python/src/ExtremeValueCopula_doc.i.in [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+See also
+--------
+Copula
+
+Examples
+--------
+Create a distribution:
+
+>>> import openturns as ot
+>>> copula = ot.ExtremeValueCopula(ot.SymbolicFunction('t', 't'))
```
Comment (dbarbier) [inline python/src/ExtremeValueCopula_doc.i.in]: Identity function does not meet above requirements.

Comment (regislebrun) [reply python/src/ExtremeValueCopula_doc.i.in]: Right, I intended to write `ot.SymbolicFunction('t', '1')`, which corresponds to the independent copula. Fixed.

**Comment by regislebrun** [comment]
ok thanks, done. I also reorganized the formulas for computePDF() and computeLogPDF() in a more readable way and added additional checks for non-smooth Pickand functions

File: lib/src/Uncertainty/Distribution/ExtremeValueCopula.cxx [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+/* Parameters constructor */
+ExtremeValueCopula::ExtremeValueCopula(const Function & pickandFunction)
+  : CopulaImplementation()
+  , pickandFunction_(pickandFunction)
+{
+  setName( "ExtremeValueCopula" );
+  // We set the dimension of the ExtremeValueCopula distribution
+  setDimension( 2 );
+  setPickandFunction(pickandFunction);
+  computeRange();
```
Comment (dbarbier) [inline lib/src/Uncertainty/Distribution/ExtremeValueCopula.cxx]: You must add
```
setParallel(false);
```
here (and in other constructors), there is no Function::setParallel method.

Comment (regislebrun) [reply lib/src/Uncertainty/Distribution/ExtremeValueCopula.cxx]: right. I have a hard time trying to fix my Python installation in order to compile the documentation, so I skip some tests from time to time, sorry.

File: lib/src/Uncertainty/Distribution/ExtremeValueCopula.cxx [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+
+/* Compute the CDF of Xi | X1, ..., Xi-1. x = Xi, y = (X1,...,Xi-1) */
+Scalar ExtremeValueCopula::computeConditionalCDF(const Scalar x, const Point & y) const
+{
+  const UnsignedInteger conditioningDimension = y.getDimension();
+  if (conditioningDimension >= getDimension()) throw InvalidArgumentException(HERE) << "Error: cannot compute a conditional CDF with a conditioning point of dimension greater or equal to the distribution dimension.";
+  // Special case for no conditioning or independent copula
+  if ((conditioningDimension == 0) || (hasIndependentCopula())) return x;
+  const Scalar u = y[0];
+  const Scalar v = x;
```
Comment (jschueller) [inline lib/src/Uncertainty/Distribution/ExtremeValueCopula.cxx]: warning: unused variable ‘v’ [-Wunused-variable]

File: python/test/t_ExtremeValueCopula_std.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    print("Quantile=", repr(quantile))
+
+    # Get 95% survival function
+    inverseSurvival = Point(copula.computeInverseSurvivalFunction(0.95))
+    print("InverseSurvival=", repr(inverseSurvival))
+    print("Survival(inverseSurvival)=%.6f" %
+          copula.computeSurvivalFunction(inverseSurvival))
+    print("entropy=%.6f" % copula.computeEntropy())
+
+    # Confidence regions
```
Comment (sofianehaddad) [inline python/test/t_ExtremeValueCopula_std.py]: could you please remove this commented block if useless ?
Btw could you add a small check for `computeConditionalCDF`, `computeConditionalQuantile` ?

Comment (jschueller) [reply python/test/t_ExtremeValueCopula_std.py]: Once Sofiane's remark is taken into account this is ok to merge.

**Comment by jschueller** [comment]
A rebase is needed though.

File: python/test/t_ExtremeValueCopula_std.py (hunk 5 above)
Comment (sofianehaddad) [reply python/test/t_ExtremeValueCopula_std.py]: I missed your comment. Maybe we could split the test into 2/3 tests  in order to override the Timeout?

**Pull request merged**